doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,674 @@
+    - ingress:
+          type: wildcard
+      egress:
+          type: interface
+          interface: 2
+      mpls_label: 3
+
+
+
+Border router
+^^^^^^^^^^^^^^^
+
+For a router to query the DRKey secret value from the control service, once has to enable this.
+
+Example (br1-ff00_0_110-1.toml)::
+
+    [router]
+    use_drkey = true
+
+Considerations for future work
+--------------------------------
+
+SCMP response
+^^^^^^^^^^^^^^^
+
+With the current implementation, the sending endhost is not being informed when his packet gets dropped due to a FABRID error.
+In the future the border routers we might think about whether to enable routers to send an SCMP response if they encounter
+an error when processing FABRID which might help the sending endhost in figuring out why his packet does not arrive at its destination.
+However, we have to be careful to not introduce additional attack vectors, e.g. for volumetric denial-of-service or reflection attacks.
+
+EPIC-HP as extension with Identifier option
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We could create a new HBH extension for EPIC hidden-path, which uses the Identifier option, instead of the
+current design which uses a SCION path-type.
+This allows the use of EPIC HP also in a incremental deployment like we have with FABRID.
+And additionally, we could also use FABRID together with EPIC HP.
+
+.. _RAINBOW:
+
+The RAINBOW system
+^^^^^^^^^^^^^^^^^^^^^
+
+The RAINBOW system allows marking traffic as higher quality of service, to be
+prioritized at on-path BRs.
+We have designed FABRID such that RAINBOW can be implemented on top of it.
+In RAINBOW, a border router reserves a certain amount of bandwidth for RAINBOW traffic, which is divided evenly between different source ASes.
+As this division happens on the level of ASes and not individual endhosts, the BR also needs to be able to authenticate traffic at an AS level.
+The FABRID HBH extension makes this possible, by including an "AS level key" flag, which specifies that the original HVF has been authenticated
+again by the corresponding AS-AS DRKey, which is only known to trusted infrastructure in the source AS.
+
+FABRID policy selection extensions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the FABRID policy selection procedure, we could add shortcuts for the identifiers, such as
+``2@REJECT``, which rejects all paths that pass through ISD2.
+
+The query mechanism could be extended to only apply policies from EXPRESSION_IF_TRUE on the hops that have matched the QUERY_EXPRESSION.
+Similarly policies from EXPRESSION_IF_FALSE would be applied to hops that do not match the QUERY_EXPRESSION.
+
+Global policy list
+^^^^^^^^^^^^^^^^^^^^^
+
+In the current implementation, we only have local policies.
+To use global policies we need a place where we can store them in a append-only fashion, that can be fetched from all ASes.
+A possible suggestion would be to create an append-only list and store it in the SCIONLab github repository.
```

Comment:
One possibility could be to create...

github -> GitHub

Reply:
Done